pgmpy/tests/test_models/test_FactorGraph.py

```diff
@@ -280,5 +280,25 @@ def test_check_model4(self):
         self.graph.add_edges_from([('a', phi3), ('c', phi3)])
         self.assertRaises(ValueError, self.graph.check_model)
 
+    def test_copy(self):
+        self.graph.add_nodes_from(['a', 'b', 'c'])
+        phi1 = DiscreteFactor(['a', 'b'], [2, 2], np.random.rand(4))
+        phi2 = DiscreteFactor(['b', 'c'], [2, 2], np.random.rand(4))
```

Comment:
@ankurankan  These two statements are present at many places but its not same everywhere . I mean if you see these in every method of the class , you will observe that there are places where the arguments are a bit different . Please correct me if i am wrong.